### DISCUSSION TITLE
Adding ref to state from action

### DISCUSSION THREAD

**Discussion by nVitius** [vuejs/pinia#673] [Help and Questions] [answered]
I've seen the doc that mentions how a plugin can add a ref to a store's state: https://pinia.esm.dev/core-concepts/plugins.html#adding-new-state.

I am trying to do the same thing but from an action instead:
```JS
type AuthStoreState = {
  user: CurrentUserQuery['currentUser'] | null
}

export const useAuthStore = defineStore('auth', {
  state: (): AuthStoreState => ({ user: null }),
  getters: {},
  actions: {
    async authenticate(email: string, password: string) {
      /* authentication code omitted for brevity */

      const { result: currentUserResult } = useCurrentUserQuery()
      const user = useResult(currentUserResult) // Ref<CurrentUserQuery['currentUser']>
      
      this.user = user // TS complains
      this.$state.user = user // TS complains here too
    }
  }
})
```

Is this just a TS issue? Is there a way to do this without having to manage the ref myself?

**Comment by posva** [comment]
Yeah, this is just a TS issue. Now that TS accepts different getters and setters for properties, this can be achieved

**Reply by posva** [reply, marked as answer]
It actually depends on https://github.com/microsoft/TypeScript/issues/43826
